File: Python/Priority Queue/PriorityQueue.py
class PriorityNode:
    # Constructors
    def __init__(self):
        self._data = 0
        self._priority = 0
        self._next = None

    def __init__(self, data):
        self._data = data
        self._priority = 0
        self._next = None

    def __init__(self, data, priority):
        self._data = data
        self._priority = priority
        self._next = None

    def __init__(self, data, next):
        self._data = data
        self._priority = 0
        self._next = next

    def __init__(self, data, priority, next):
        self._data = data
        self._priority = priority
        self._next = next


class PriorityQueue:
    # Constructors
    def __init__(self):
        self._front = None
        self._back = None
        self._size = 0

    def __init__(self, data):
        self._front = PriorityNode(data)
        self._back = self._front
        self._size = 1

    def __init__(self, list):
        self._front = PriorityNode(list[0])
        self._back = self._front
        self._size = 1

        for i in range(1, len(list)):
            self._back._next = PriorityNode(list[i])
            self._back = self._back._next
            self._size += 1

    # Methods
    def enqueue(self, data, priority):
        if self.is_empty():
            self._front = PriorityNode(data, priority)
            self._back = self._front
            self._size += 1
            return

        if priority < self._front._priority:
            self._front = PriorityNode(data, priority, self._front)
            self._size += 1
            return

        current = self._front
        while current._next != None and current._next.priority <= priority:
            current = current._next

        current._next = PriorityNode(data, priority, current._next)
        self._size += 1

    def dequeue(self):
        if self.is_empty():
            print("Queue is empty")
            return

        self._front = self._front._next
        self._size -= 1

    def peek(self):
        if self.is_empty():
            print("Queue is empty")
            return -1

        return self._front._data

    def is_empty(self):
        return self._size == 0

    def get_size(self):
        return self._size

File: C#/Priority Queue/Priority_Queue.cs
public class Priority_Node
{
    #region Properties
    private int _data;
    private int _priority;
    private Priority_Node _next;
    #endregion

    #region Constructors
    public Priority_Node()
    {
        this._data = 0;
        this._priority = 0;
        this._next = null;
    }

    public Priority_Node(int data)
    {
        this._data = data;
        this._priority = 0;
        this._next = null;
    }

    public Priority_Node(int data, int priority)
    {
        this._data = data;
        this._priority = priority;
        this._next = null;
    }

    public Priority_Node(int data, Priority_Node next)
    {
        this._data = data;
        this._priority = 0;
        this._next = next;
    }

    public Priority_Node(int data, int priority, Priority_Node next)
    {
        this._data = data;
        this._priority = priority;
        this._next = next;
    }
    #endregion
}

public class Priority_Queue
{
    #region Properties
    private Priority_Node _front;
    private Priority_Node _back;
    private int _size;
    #endregion

    #region Constructors
    public Priority_Queue()
    {
        this._front = null;
        this._back = null;
        this._size = 0;
    }

    public Priority_Queue(int data)
    {
        this._front = new Priority_Node(data);
        this._back = this._front;
        this._size = 1;
    }

    public Priority_Queue(int[] arr)
    {
        this._front = new Priority_Node(arr[0]);
        this._back = this._front;
        this._size = 1;
        for (int i = 1; i < arr.Length; i++)
        {
            this._back._next = new Priority_Node(arr[i]);
            this._back = this._back._next;
            this._size++;
        }
    }
    #endregion

    #region Methods
    public void enqueue(int data, int priority)
    {
        if (this._size == 0)
        {
            this._front = new Priority_Node(data, priority);
            this._back = this._front;
            this._size = 1;
            return;
        }

        Priority_Node temp = this._front;
        Priority_Node prev = null;
        while (temp != null && temp._priority >= priority)
        {
            prev = temp;
            temp = temp._next;
        }

        if (prev == null)
        {
            this._front = new Priority_Node(data, priority, this._front);
            this._size++;
            return;
        }

        prev._next = new Priority_Node(data, priority, temp);
        this._size++;
    }

    public void dequeue()
    {
        if (this._size == 0)
        {
            print("Queue is empty");
            return;
        }

        this._front = this._front._next;
        this._size--;
    }

    public int peek()
    {
        if (this._size == 0)
        {
            print("Queue is empty");
            return -1;
        }

        return this._front._data;
    }

    public int get_size()
    {
        return this._size;
    }
    #endregion
}